### ISSUE TITLE
Dynamic Things and semantic vocabularies (`@context`)

### ISSUE THREAD

**Issue by mkovatsc** [w3c/wot-scripting-api#84]
Thinking a bit more about implementation aspects for handling semantics, I came to the following conclusions (for now):

* The API for semantic annotations it quite clumsy at the moment
* Dynamically changing the vocabulary of a TD introduces a lot of complexity for handling Things (i.e., is bad) -- often WoT Client must rely on a fixed set of vocabularies that are implemented (no dynamic loading and reasoning)
* Might be best to limit declaration of @context entries, and hence prefixes and possible annotations to the `WoT.expose()` function
* Using a different set of @context entries would need the creation of a new ExposedThing object -- as it is also a fundamental change in the Thing

**Comment by zolkis** [comment]
I definitely agree with this thinking. Any change proposals in the API itself, or the changes should apply to the algorithm definitions?

I would offload the scripting API from semantics, TD vocabularies etc. My current understanding is that the scripting use cases are doable with TD related data regarded as opaque strings from scripts.

That is, we can expose a thing if a TD is provided to the script (as an opaque string) - the implementation will know how to parse, identify versions, map to protocol bindings, etc.

Some things we could change from scripts - the question is what is the minimum amount for our use cases.

**Comment by mkovatsc** [comment]
Well, not fully decoupled, which would void the programmatic approach to Things and their TDs.

In the current proposal, see 
```javascript
dictionary SemanticType {
    DOMString name;
    USVString context;
    DOMString prefix;
};
```
and
```javascript
...
sequence<SemanticType>? semanticTypes = [];
...
```
in the `Thing*Init` dictionaries.

**Comment by zolkis** [comment]
Should be fixed by #113.
